Add lsp-ivy-show-symbol-kind and lsp-ivy-filter-symbol-kind options

orausch opened the pull request:
These options show the type of a symbol in the ivy completion. I found these extensions useful for myself, feel free to merge them if you think it's a good fit for the package.

Here's a screenshot of the default configuration:
![image](https://user-images.githubusercontent.com/22137236/77010954-1d2b1000-696b-11ea-9396-17defaea796f.png)

sebastiansturm commented:
thank you, that looks nice! Will be using this the next few days and merge it if I don't notice any bugs

sebastiansturm commented:
sorry for being so slow to respond. I have now tried out your PR; works very well, looks much prettier than the old implementation and I love being able to specifically search for functions, enums, etc. using `lsp-ivy-filter-symbol-kind`. One minor nitpick -- `lsp-ivy-show-symbol-kind` doesn't do anything, maybe you forgot to use it in `lsp-ivy--format-symbol-match`? Personally I'd also be fine with not making this a user-configurable option, unless the extra labels cause a perceptible slowdown (at least on my machine they don't, as far as I can tell)

orausch commented:
Oops, you're right. I added a check for that in `lsp-ivy--format-symbol-match`. 

>  unless the extra labels cause a perceptible slowdown (at least on my machine they don't, as far as I can tell)

Yep, I didn't notice any either, but I added the option to disable it for that reason.

orausch commented:
Glad to hear that it works by the way! Feel free to let me know if there are any issues with the changes; I'm not that experienced with elisp.

sebastiansturm commented:
I'm not an elisp expert either (far from it), but I guess the PR is fine. Undoing the formatting changes is not really necessary, but if you don't mind doing so, I'll gladly take the offer :)

orausch commented:
All done

sebastiansturm commented:
thanks!
